helm/idol-nifi/templates/nifi/nifi.yaml

```diff
@@ -0,0 +1,204 @@
+        envFrom:
+          - configMapRef:
+              name: nifi-cm
+              optional: false
+        volumeMounts:
+          - name: data
+            mountPath: /opt/nifi/nifi-current/data
+            subPath: data
+            readOnly: false
+          - name: data
+            mountPath: /opt/nifi/nifi-current/keytool
+            subPath: keytool
+            readOnly: false
+          - name: scripts
+            mountPath: /scripts
+            readOnly: false
+        lifecycle:
+          postStart:
+            exec:
+              command:
+              - /bin/bash
+              - -c
+              - |
+                logfile=/opt/nifi/nifi-current/logs/post-start.log
+                /scripts/security.sh | tee -a ${logfile}
+                /scripts/wait.sh | tee -a ${logfile}
+                /scripts/connect-registry.sh | tee -a ${logfile}
+                echo [$(date)] postStart completed | tee -a ${logfile}
+          preStop:
+            exec:
+              command:
+              - sh
+              - -c
+              - |
+                host=nifi-0.${NIFI_WEB_HTTP_HOST#*.}
+                port=$NIFI_WEB_HTTP_PORT
+                cluster_node_id=`curl -s http://$NIFI_WEB_HTTP_HOST:$NIFI_WEB_HTTP_PORT/nifi-api/controller/cluster | jq -r ".cluster.nodes[] | select(.address==\"$NIFI_WEB_HTTP_HOST\") | .nodeId "`
+
+                setAndWaitForStatus () {
+                    curl -s -X PUT http://$host:$port/nifi-api/controller/cluster/nodes/$cluster_node_id -H 'Content-Type: application/json' -d "{\"node\":{\"nodeId\":\"$cluster_node_id\",\"status\": \"$1\"}}"
+                    cluster_node_status="$1"
+                    while [ "$cluster_node_status" != "$2" ]
+                    do
+                        sleep 1
+                        cluster_node_status=`curl -s http://$host:$port/nifi-api/controller/cluster/nodes/$cluster_node_id | jq .node.status -r`
+                    done
+                }
+
+                if [ ! -z "$cluster_node_id" ]
+                then
+                    setAndWaitForStatus "DISCONNECTING" "DISCONNECTED"
+                    setAndWaitForStatus "OFFLOADING" "OFFLOADED"
+                    curl -s -X DELETE http://$host:$port/nifi-api/controller/cluster/nodes/$cluster_node_id
+                fi
+        livenessProbe:
+          exec:
+            command:
+              - pgrep
+              - java
+          initialDelaySeconds: 60
+          periodSeconds: 30
+          timeoutSeconds: 10
+          failureThreshold: 3
+          successThreshold: 1
+        readinessProbe:
+          tcpSocket:
+            port: cluster
+          initialDelaySeconds: 30
+          periodSeconds: 30
+          timeoutSeconds: 10
+          failureThreshold: 3
+          successThreshold: 1
+        resources:
+          requests:
+            cpu: 400m
+            memory: 2Gi
+          limits:
+            cpu: 400m
+            memory: 2Gi
```

Comment:
These should be configurable and probably higher default values.
See also NIFI_JVM_HEAP_INIT/NIFI_JVM_HEAP_MAX in the `configmap.yaml`

Reply:
[438faf0](https://github.com/opentext-idol/idol-containers-toolkit/pull/12/commits/438faf01f6eb7f7d02fe5f2c5ae42a0874f81af8)